ros_ws/src/autonomous/src/emergency_stop.cpp

```diff
@@ -10,32 +10,32 @@ EmergencyStop::EmergencyStop()
 bool EmergencyStop::emergencyStop(const sensor_msgs::LaserScan::ConstPtr& lidar)
 {
     int available_samples = (lidar->angle_max - lidar->angle_min) / lidar->angle_increment;
-    int samples_used = 0;
-    float range_sum = 0;
 
-    int index_start = available_samples / 2 - SAMPLE_ANGLE / lidar->angle_increment / 2;
-    int index_end = available_samples / 2 + SAMPLE_ANGLE / lidar->angle_increment / 2;
+    // calculate the sample_angle of the lidar to be used to check if there is
+    // an obstacle in front of the car, with respect to the RANGE_THRESHOLD
+    float sample_angle = (360 * CAR_BUMPER_LENGTH) / (M_PI * RANGE_THRESHOLD * 2) * DEG_TO_RAD;
```

Comment:
What is the reasoning behind calculating the `sample_angle` like this? I expected something like `atan(CAR_BUMPER_LENGTH / 2 / RANGE_THRESHOLD)`.
Also, I think that the comment above the line isn't needed, since it explains what is done, but not why.

Reply:
I wanted the arc length of the lidar to be the same as the width of the car. Since the arc length is dependent on the arc angle (sample_angle) and the radius (RANGE_THRESHOLD), I used this [formula](https://theengineeringmindset.com/arc-length-how-to-calculate/) to determine the exact angle, at which the arc length will be the same as the width of the car, with a radius of RANGE_THRESHOLD.